articles/client-platforms/react/05-linking-accounts.md

```diff
@@ -0,0 +1,372 @@
+
+```javascript
+/* ===== ./src/utils/AuthService.js ===== */
+import { EventEmitter } from 'events'
+import Auth0Lock from 'auth0-lock'
+
+export default class AuthService extends EventEmitter {
+  constructor(clientId, domain) {
+    super()
+    this.clientId = clientId
+    this.domain = domain
+    // Configure Auth0
+    this.lock = new Auth0Lock(clientId, domain)
+    // Add callback for lock `authenticated` event
+    this.lock.on('authenticated', this._doAuthentication.bind(this))
+    // Add callback for lock `authorization_error` event
+    this.lock.on('authorization_error', this._authorizationError.bind(this))
+    // binds login functions to keep this context
+    this.login = this.login.bind(this)
+  }
+
+  _doAuthentication(authResult){
+    authResult.state = authResult.state || '' //making sure state exists
+    if (authResult.state.includes('linking')){
+      this.linkAccount(authResult.idToken) // linkAccount when state is linking
+    } else {
+      // Otherwise saves the user token
+      this.setToken(authResult.idToken)
+      // Async loads the user profile data
+      this.lock.getProfile(authResult.idToken, (error, profile) => {
+        if (error) {
+          console.log('Error loading the Profile', error)
+        } else {
+          this.setProfile(profile)
+        }
+      })
+    }
+  }
+
+  ... // omitting methods
+
+  fetchApi(url, options){
+    // performs api calls sending the required authentication headers
+    const headers = {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+      'Authorization': 'Bearer ' + this.getToken()
+    }
+
+    const userId = this.getProfile().user_id
+    return fetch(`<%= "https://${this.domain}/api/v2/users/${userId}/${url}" %>`, {
+      headers,
+      ...options
+    })
+    .then(response => response.json())
+  }
+
+  linkAccount(token){
+    // prepares api request body data
+    const data = {
+      link_with: token
+    }
+    // sends a post to auth0 api to create a new identity
+    return this.fetchApi('identities', {
+      method: 'POST',
+      body: JSON.stringify(data)
+    })
+    .then(identities => {
+      const profile = this.getProfile()
+      this.setProfile({...profile, identities}) // updates profile identities
+    })
+  }
+}
+```
+
+The code introduces also two new methods: `fetchApi` to send requests to auth0 users api with the required headers and parsing the requests to json. Finally, `linkAccount` uses `fetchApi` to send a _POST_ request and create a new identity in user account, updating the stored profile after a successful response. For more details, check the [user identities endpoint documentation](https://auth0.com/docs/api/management/v2#!/Users/post_identities).
```

Comment:
@flaviogranero 
Update the link to: `/api/management/v2#!/Users/post_identities`